pkg/backup/backup_test.go

```diff
@@ -596,6 +596,164 @@ func TestBackupResourceFiltering(t *testing.T) {
 	}
 }
 
+// TestCRDInclusion tests whether related CRDs are included, based on
+// backed-up resources and "include cluster resources" flag, and
+// verifies that the set of items written to the backup tarball are
+// correct. Validation is done by looking at the names of the files in
+// the backup tarball; the contents of the files are not checked.
+func TestCRDInclusion(t *testing.T) {
+	tests := []struct {
+		name         string
+		backup       *velerov1.Backup
+		apiResources []*test.APIResource
+		want         []string
+	}{
+		{
+			name: "include cluster resources=auto includes all CRDs when backing up all namespaces",
```

Comment:
do you mean full-cluster backup? all namespaces is a little confusing since non-namespaced resources are also backed up